cmd/exo/cmd/privnet_show.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/exoscale/egoscale"
+	"github.com/exoscale/egoscale/cmd/exo/table"
+	"github.com/spf13/cobra"
+)
+
+// showCmd represents the show command
+var privnetShowCmd = &cobra.Command{
+	Use:   "show <privnet name | id>",
+	Short: "Show a private network details",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			return cmd.Usage()
+		}
+		network, vms, err := privnetDetails(args[0])
+		if err != nil {
+			return err
+		}
+
+		table := table.NewTable(os.Stdout)
+		table.SetHeader([]string{"Zone", "Name", "Virtual Machine", "Virtual Machine ID"})
+		zone := network.ZoneName
+		name := network.Name
+		if len(vms) > 0 {
+			for _, vm := range vms {
+				table.Append([]string{zone, name, vm.Name, vm.ID})
+				zone = ""
+				name = ""
+			}
+		} else {
+			table.Append([]string{zone, network.Name, "", ""})
+		}
+		table.Render()
+
+		return nil
+	},
+}
+
+func privnetDetails(privnetName string) (*egoscale.Network, []egoscale.VirtualMachine, error) {
+
+	resp, err := cs.List(&egoscale.Zone{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	network, err := searchPrivnet(privnetName, resp)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
I don't get why you're doing O(n) calls instead of O(1) here? It's a bit more work to filter the networks and create a `map[zone]network` but it should be worth it.

I would inform the user if more than one network was found for a given name for any show/delete operation